lib/quicly.c

```diff
@@ -1622,6 +1677,8 @@ static int commit_send_packet(quicly_conn_t *conn, struct st_quicly_send_context
 {
     assert(s->target.cipher->aead != NULL);
 
+    assert(s->dst != s->dst_encrypt_from);
```

Comment:
What does this assertion check for?

Reply:
It checks that we are not trying to commit a packet that does not have any payload. This should be guaranteed by `prepare_packet` when it allocates a buffer for a packet.